src/app/view.component.ts

```diff
@@ -510,7 +511,54 @@ export class ViewComponent implements OnInit, AfterViewInit {
       obj = obj[key_location].children;
     }
   }
+  /* This function is used to load Precipitation Data in oneclick from TopDataset Button
+   */
+  LoadPrecipitation() {
+    const selectedDataset = this._model.settings.Datasets.find(myObj =>
+      myObj.FullName.includes("Precipitation|Single Level|Monthly Mean")
+    );
+    this._controller.loadLevels(selectedDataset);
+    this._controller.loadDataset(selectedDataset, selectedDataset.StartDate, 1);
+  }
+  /* This function is used to load Air Temperature Data in oneclick from TopDataset Button
+   */
+  LoadAirTemperature() {
```

Comment:
public/private?

Reply:
made it private as it is called within its component